sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/ReplaceFeatureModifier.cs

```diff
@@ -0,0 +1,193 @@
+﻿namespace Mapbox.Unity.MeshGeneration.Modifiers
+{
+	using System.Collections.Generic;
+	using UnityEngine;
+	using Mapbox.Unity.MeshGeneration.Data;
+	using System;
+	using Mapbox.Unity.Map;
+	using Mapbox.Utils;
+	using Mapbox.Unity.Utilities;
+	using Mapbox.VectorTile.Geometry;
+	using Mapbox.Unity.MeshGeneration.Interfaces;
+
+
+	/// <summary>
+	/// ReplaceBuildingFeatureModifier takes in POIs and checks if the feature layer has those points and deletes them
+	/// </summary>
+	[CreateAssetMenu(menuName = "Mapbox/Modifiers/Replace Feature Modifier")]
+	public class ReplaceFeatureModifier : GameObjectModifier, IReplacementCriteria
+	{
+		private List<string> _latLonToSpawn;
+
+		private Dictionary<GameObject, GameObject> _objects;
+		[SerializeField]
+		private SpawnPrefabOptions _options;
+		private List<GameObject> _prefabList = new List<GameObject>();
+
+		[SerializeField]
+		[Geocode]
+		private List<string> _prefabLocations;
+
+		private string _featureId;
+
+		public override void Initialize()
+		{
+			base.Initialize();
+			//duplicate the list of lat/lons to track which coordinates have already been spawned
+			_latLonToSpawn = new List<string>(_prefabLocations);
+			_featureId = String.Empty;
+			if (_objects == null)
+			{
+				_objects = new Dictionary<GameObject, GameObject>();
+			}
+			_latLonToSpawn = new List<string>(_prefabLocations);
+		}
+
+		public override void SetProperties(ModifierProperties properties)
+		{
+			_options = (SpawnPrefabOptions)properties;
+		}
+
+		/// <summary>
+		/// Check the feature against the list of lat/lons in the modifier
+		/// </summary>
+		/// <returns><c>true</c>, if the feature overlaps with a lat/lon in the modifier <c>false</c> otherwise.</returns>
+		/// <param name="feature">Feature.</param>
+		public bool ShouldReplaceFeature( VectorFeatureUnity feature )
+		{
+			foreach( var point in _prefabLocations )
+			{
+				var coord = Conversions.StringToLatLon(point);
+				if (feature.ContainsLatLon(coord))
+				{
+
+
+					if(feature.Data.Id != null && !string.IsNullOrEmpty(feature.Data.Id.ToString()) && String.IsNullOrEmpty(_featureId))
```

Comment:
`Id` is of type [`ulong`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/ulong) which has [a default value of `0`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/default-values-table).

This means that the first two conditions in this `if` clause will always evalute to `true`.

Please fix.